src/friendly-iframe-embed.js

```diff
@@ -267,6 +268,12 @@ function mergeHtml(spec) {
     });
   }
 
+  // Load CSP
```

Comment:
So we don't necessarily have to fix it in this PR, but I don't understand exactly how this method's ad-hoc HTML parsing works, and for a security measure specifically designed to defend against XSS, that makes me nervous. I'd be interested in seeing if we could switch to the "paranoid" version I proposed on an internal thread (require every signed creative to start with `<!doctype html><html amp4ads><head><meta charset=utf-8>`).

Reply:
Right now if the HTML is invalid, the CSP would be inserted before the invalid text so html of 'foo' would result in '<meta http-equiv=Content-Security-Policy...>foo'.  Likely the browser would do the right thing in that scenario so I think we can leave it as is.